tests/vanilla/unstable_derive.test.tsx

```diff
@@ -84,16 +82,96 @@ describe('unstable_derive for scoping atoms', () => {
     expect(derivedStore.get(c)).toBe('ab')
 
     derivedStore.set(a, 'a2')
-    await new Promise((resolve) => setTimeout(resolve))
     expect(store.get(c)).toBe('ab')
     expect(derivedStore.get(c)).toBe('a2b')
   })
 
+  /**
+   * a, b(a)
+   * S1[a]: a1, b0(a1)
+   */
+  it('derived atom with subscribe', () => {
+    const a = atom('a')
+    const b = atom(
+      (get) => get(a),
+      (_get, set, v: string) => set(a, v),
+    )
+    const scopedAtoms = new Set<Atom<unknown>>([a])
+
+    function makeStores() {
+      const store = createStore()
+      const derivedStore = store.unstable_derive(
+        (getAtomState, atomRead, atomWrite) => {
+          const scopedAtomStateMap = new WeakMap()
+          return [
+            (atom, originAtomState) => {
+              if (scopedAtoms.has(atom)) {
+                let atomState = scopedAtomStateMap.get(atom)
+                if (!atomState) {
+                  atomState = { d: new Map(), p: new Set(), n: 0 }
+                  scopedAtomStateMap.set(atom, atomState)
+                }
+                return atomState
+              }
+              return getAtomState(atom, originAtomState)
+            },
+            atomRead,
+            atomWrite,
+          ]
+        },
+      )
+      expect(store.get(b)).toBe('a')
+      expect(derivedStore.get(b)).toBe('a')
+      return { store, derivedStore }
+    }
+
+    /**
+     * Ba[ ]: a0, b0(a0)
+     * S1[a]: a1, b0(a1)
+     */
+    {
+      const { store, derivedStore } = makeStores()
+      store.set(b, '*')
+      expect(store.get(b)).toBe('*')
+      expect(derivedStore.get(b)).toBe('a')
+    }
+    {
+      const { store, derivedStore } = makeStores()
+      derivedStore.set(b, '*')
+      expect(store.get(b)).toBe('a')
+      expect(derivedStore.get(b)).toBe('*')
+    }
+    {
+      const { store, derivedStore } = makeStores()
+      const storeCallback = vi.fn()
+      const derivedCallback = vi.fn()
+      store.sub(b, storeCallback)
+      derivedStore.sub(b, derivedCallback)
+      store.set(b, '*')
+      expect(store.get(b)).toBe('*')
+      expect(derivedStore.get(b)).toBe('a') // FIXME: received '*'
```

Comment:
I think it's because when subscribed, the store updates mounted atoms eagerly, so it makes sense to fail.
But, you made a small problem that I can understand. I will tackle it later.